src/functions.js

```diff
@@ -128,6 +171,17 @@ function factorial( int ) {
 
   // calculate and return the factorial of int
   // note: factorial of 0 is 1
+
+  if ( int === 0) {
+    return 1;
+  } else if( int < 0) {
+    return undefined;
+  } else {
+    for ( var i = int; --i; ) {
```

Comment:
Are you sure you completely understood how it works ? It seems quite difficult to me...
Give a look at how recursion work with [this example](https://www.dyclassroom.com/recursion-algorithm/factorial) on the factorial and maybe try to implement this function with recursion 😉